server/routes/caseNotes/caseNotesController.ts

```diff
@@ -46,4 +56,64 @@ export default class CaseNotesController {
     const view = new CaseNotesView(presenter)
     ControllerUtils.renderWithLayout(res, view, null)
   }
+
+  async startAddCaseNote(
+    req: Request,
+    res: Response,
+    loggedInUserType: 'service-provider' | 'probation-practitioner'
+  ): Promise<void> {
+    const draftCaseNote = await this.draftsService.createDraft('caseNote', null, {
+      userId: res.locals.user.userId,
+    })
+
+    res.redirect(`/${loggedInUserType}/referrals/${req.params.id}/add-case-note/${draftCaseNote.id}/details`)
+  }
+
+  async addCaseNote(
+    req: Request,
+    res: Response,
+    loggedInUserType: 'service-provider' | 'probation-practitioner'
+  ): Promise<void> {
+    const { accessToken } = res.locals.user.token
+    const referralId = req.params.id
+    const fetchResult = await this.fetchDraftCaseNoteOrRenderMessage(req, res)
+    if (fetchResult.rendered) {
+      return
+    }
+    const draftCaseNote = fetchResult.draft
+    let error: FormValidationError | null = null
+    let userInputData: Record<string, string> | null = null
+    if (req.method === 'POST') {
+      const data = await new AddNewCaseNoteForm(req).data(referralId)
+      if (!data.error) {
+        try {
+          await this.draftsService.updateDraft(draftCaseNote.id, data.paramsForUpdate, {
```

Comment:
I might be misunderstanding here, but do we need to update the draft here if we're finalising the case note on the next line?

Reply:
Yes, technically a draft referral is not necessary at all with a single page.
It comes into play when there is more than one page on the journey. 

I must have added it into the wrong commit, as it makes more sense in the next commit (on another PR) where there is a check your answers page.

I've removed it - I will make sure the following PR is rebased after this is merged into main.